src/views/RegisterPage/__tests__/RegisterPage.test.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import expect from 'expect';
+
+import {
+  mapStateToProps,
+  RegisterPage,
+} from '../RegisterPage';
+
+
+const setUp = () => {
+  const props = {
+    handleSubmit: jest.fn(),
+    signUpAction: jest.fn(),
+  };
+  const wrapper = mount(<RegisterPage {...props} />);
+
+  return {
+    props,
+    wrapper,
+  };
+};
+
+describe('Register page Rendered: ', () => {
+  const { wrapper } = setUp();
+  it(' render header component', () => {
+    const gridMenu = wrapper.find('.register');
+
+    expect(gridMenu.exists())
+      .toBe(true);
+  });
+
+  it(' render form component', () => {
+    const formFile = wrapper.find('.register-form');
+
+    expect(formFile.exists())
+      .toBe(true);
+  });
+
+  it(' test handle onsubmit event', () => {
+    wrapper.instance()
+      .handleSubmit({
+        preventDefault() {
+        },
+      });
+    const expectedState = {
+      email: '',
+      password: '',
+      username: '',
+    };
+    expect(wrapper.state())
+      .toEqual(expectedState);
+  });
+
+  it(' should handle onchange function', () => {
+    wrapper
+      .instance()
+      .handleInputChange({
+        target: {
+          name: 'email',
+          value: 'test@gmail.com',
+        },
+      });
+    wrapper
+      .instance()
+      .handleInputChange({
+        target: {
+          name: 'username',
+          value: 'test',
+        },
+      });
+    wrapper
+      .instance()
+      .handleInputChange({
+        target: {
+          name: 'password',
+          value: 'test@Gmail',
+        },
+      });
+    expect(wrapper.state('email'))
```

Comment:
When testing the frontend, we're supposed to focus on testing behavior from the user's point of view rather than testing implementation (i.e `handleInputChange` method being called). So, for example in the test above, instead of testing whether the data got to the state, you should submit the form and assert that the user sees either a success/failure message.